src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,262 @@
+     */
+    public static Event validateEvent(String req) throws NoDescriptionException, MissingTimeCommandException {
+        if (req.equals("event")) {
+            throw new NoDescriptionException("The description of an event cannot be empty.");
+        }
+        if (!req.contains("/at")) {
+            throw new MissingTimeCommandException("Missing Time Command: add '/at' in the command.");
+        }
+
+        String[] splitReq = req.split(" ", 2);
+        String[] body = splitReq[1].split(" /at ", 2);
+        String desc = body[0];
+        String date = body[1];
+        Event event = new Event(desc, date);
+        return event;
+    }
+
+    /**
+     * Validates a done command.
+     *
+     * Static class method.
+     *
+     * Receives an index to specify a task in the TaskList.
+     * Converts input string to integer.
+     *
+     * @param req The full command
+     * @return Integer index of the task in the task list
+     * @throws NoDescriptionException If no description is appended after command.
+     * @throws InvalidDescriptionException If description after command is not a valid index.
+     * @throws IOException
+     */
+    public static int validateDone(String req) throws NoDescriptionException, InvalidDescriptionException,
+            IOException {
+        if (req.equals("done")) {
+            throw new NoDescriptionException("Please specify the task number.");
+        }
+
+        String[] splitReq = req.split(" ", 2);
+        String desc = splitReq[1];
+
+        int index;
+        try {
+            index = Integer.parseInt(desc);
+        } catch (NumberFormatException e) {
+            throw new InvalidDescriptionException("Please append a task number after 'done'.");
+        }
+
+        return index;
+    }
+
+    /**
+     * Validates a delete command.
+     *
+     * Receives an index to specify a task to delete.
+     * Converts input string to integer.
+     *
+     * @param req The full command
+     * @return Integer index of the task in the task list
+     * @throws NoDescriptionException If no description is appended after command.
+     * @throws InvalidDescriptionException If description after command is not a valid index.
+     * @throws IOException
+     */
+    public static int validateDelete(String req) throws NoDescriptionException, InvalidDescriptionException {
+        if (req.equals("delete")) {
+            throw new NoDescriptionException("Please specify a task to delete.");
+        }
+
+        String[] splitReq = req.split(" ", 2);
+        String desc = splitReq[1];
+
+        int index;
+        try {
+            index = Integer.parseInt(desc);
+        } catch (NumberFormatException e) {
+            throw new InvalidDescriptionException("Please append a task number after 'delete'.");
+        }
+
+        return index;
+    }
+
+    /**
+     * Validates a find command.
+     *
+     * Receives a string to match.
+     * @param req
```

Comment:
Could add description for parameter for consistency since your other method headers have param descriptions